From: SJ Park <sj@kernel.org>
Date: Sat, 8 Aug 2026 16:05:32 -0700
Subject: [PATCH] Revert "samples/damon/mtier: error out for zero quota goal
 target values"

This reverts commit a16fd3ad9d89b05475864da97327870464611736.

Now the core layer handles the corner case.

Signed-off-by: SJ Park <sj@kernel.org>
---
 samples/damon/mtier.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/samples/damon/mtier.c b/samples/damon/mtier.c
index bea45c87cc9be..27dc88bdf7a0e 100644
--- a/samples/damon/mtier.c
+++ b/samples/damon/mtier.c
@@ -161,9 +161,6 @@ static struct damon_ctx *damon_sample_mtier_build_ctx(bool promote)
 	if (!scheme)
 		goto free_out;
 	damon_set_schemes(ctx, &scheme, 1);
-	/* zero target value causes division by zero in damos_quota_store() */
-	if (!node0_mem_used_bp || !node0_mem_free_bp)
-		goto free_out;
 	quota_goal = damos_new_quota_goal(
 			promote ? DAMOS_QUOTA_NODE_MEM_USED_BP :
 			DAMOS_QUOTA_NODE_MEM_FREE_BP,
-- 
2.47.3

